src/coap_client_zephyr.c

```diff
@@ -1320,3 +1302,76 @@ struct golioth_client *golioth_client_create(const struct golioth_client_config
     golioth_client_destroy(new_client);
     return NULL;
 }
+
+static void purge_request_mbox(golioth_mbox_t request_mbox)
+{
+    golioth_coap_request_msg_t request_msg = {};
+    size_t num_messages = golioth_mbox_num_messages(request_mbox);
+
+    for (size_t i = 0; i < num_messages; i++)
+    {
+        assert(golioth_mbox_recv(request_mbox, &request_msg, 0));
```

Comment:
Same note about wrapping function calls in `assert()`
```suggestion
        bool ok = golioth_mbox_recv(request_mbox, &request_msg, 0);
        assert(ok);
```